xarray/core/indexes.py

```diff
@@ -1480,13 +1480,15 @@ def _id_index(self) -> dict[int, T_PandasOrXarrayIndex]:
     @property
     def _id_coord_names(self) -> dict[int, tuple[Hashable, ...]]:
         if self.__id_coord_names is None:
-            id_coord_names: Mapping[int, list[Hashable]] = defaultdict(list)
-            for k, v in self._coord_name_id.items():
-                id_coord_names[v].append(k)
-            self.__id_coord_names = {k: tuple(v) for k, v in id_coord_names.items()}
-
+            self._create_id_coord_names()
         return self.__id_coord_names
 
+    def _create_id_coord_names(self) -> None:
```

Comment:
its strange that the function `_create_id_coord_names` gets hit all the time by the profiler. I would have thought that once the dataset is created this function would only be called once.